src/services/codefixes/importFixes.ts

```diff
@@ -620,4 +666,73 @@ namespace ts.codefix {
         // Need `|| "_"` to ensure result isn't empty.
         return !isStringANonContextualKeyword(res) ? res || "_" : `_${res}`;
     }
+
+    function createLazyPackageJsonDependencyReader(fromFile: SourceFile, host: LanguageServiceHost) {
+        const packageJsonPaths = findPackageJsons(getDirectoryPath(fromFile.fileName), host);
+        const dependencyIterator = readPackageJsonDependencies();
+        let seenDeps: Map<true> | undefined;
+        function *readPackageJsonDependencies() {
+            type PackageJson = Record<typeof dependencyKeys[number], Record<string, string> | undefined>;
+            const dependencyKeys = ["dependencies", "devDependencies", "optionalDependencies"] as const;
+            for (const fileName of packageJsonPaths) {
+                const content = readJson(fileName, { readFile: host.readFile ? host.readFile.bind(host) : sys.readFile }) as PackageJson;
+                for (const key of dependencyKeys) {
+                    const dependencyHash = content[key];

```

Comment:
what does hash refer to here?

Reply:
It’s the object specified by `dependencies`, `devDependencies`, or `optionalDependencies`. Plural names conjure the impression of arrays for me, so I wanted to make it clear that it’s an object, but I’m not attached to it.

Reply:
oh, I'd call it dependencyObject in that case.

Reply:
Renamed `dependencies` already, its typed as a record so I think it’s fine.